internal/access/credential.go

```diff
@@ -23,11 +24,26 @@ func CreateCredential(c *gin.Context, user models.Identity) (string, error) {
 		return "", HandleAuthErr(err, "user", "create", models.InfraAdminRole)
 	}
 
-	tmpPassword, err := generate.CryptoRandom(12, generate.CharsetPassword)
+	settings, err := data.GetSettings(db)
+	if err != nil {
+		return "", err
+	}
+
+GENERATE:
+	tmpPassword, err := generate.CryptoRandom(settings.LengthMin, generate.CharsetPassword)
 	if err != nil {
 		return "", fmt.Errorf("generate: %w", err)
 	}
 
+	err = checkPasswordRequirements(db, tmpPassword)
+	var validateError validate.Error
+	if err != nil {
+		if errors.As(err, &validateError) {
+			goto GENERATE
+		}
+		return "", err
+	}
```

Comment:
there's a non-zero chance that this loops forever. eg if the sum of the min counts is higher than the number of characters in the string. we should have a max number of tries and error out as a simple solution. a more complex solution would be to generate a string with exactly the characteristics requested.